### PR TITLE
qdma xvc_pri, 64bit BAR, p2p_enable ioctl

### PR DESCRIPTION
**Pull request by houlz0507**

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

File: src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c [hunk 1]
```diff
@@ -783,3 +795,52 @@ void reset_notify_client_ctx(struct xocl_dev *xdev)
...
+
+	enable = ((struct drm_xocl_p2p_enable *)data)->enable;
+
+	p2p_bar = xocl_get_p2p_bar(xdev);
+	if (p2p_bar < 0) {
+		xocl_err(&pdev->dev, "p2p bar is not configurable");
+		ret = -EINVAL;
+		goto failed;
+	}
+
```
Comment (maxzhen) [inline src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c]: I think you should be able to skip below operation if the current state is the same as what user is asking for...

File: src/runtime_src/driver/include/xclhal2.h [hunk 2]
```diff
@@ -127,6 +127,7 @@ struct xclDeviceInfo2 {
   unsigned short mVccIntVol;
   unsigned short mVccIntCurr;
   unsigned short mNumCDMA;
+  bool mP2pEnabled;
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/driver/include/xclhal2.h]: removed

File: src/runtime_src/driver/include/xclhal2.h [hunk 3]
```diff
@@ -127,6 +127,7 @@ struct xclDeviceInfo2 {
...
   // More properties here
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/driver/include/xclhal2.h]: removed

File: src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_drv.c [hunk 4]
```diff
@@ -72,6 +91,258 @@ void user_pci_reset_done(struct pci_dev *pdev)
...
+
+#endif
+
+void xocl_p2p_mem_release(struct xocl_dev *xdev, bool recov_bar_sz)
+{
+	struct pci_dev *pdev = xdev->core.pdev;
+	int p2p_bar = -1;
+
+	if (xdev->bypass_bar_addr) {
+		devres_release_group(&pdev->dev, xdev->p2p_res_grp);
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_drv.c]: did not change dem_mmremap_pages() logic. devres_release_group() is very old interface.

File: src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c (hunk 1 above)
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c]: return EACCES if error

**Pull request merged**